src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3767,6 +3779,34 @@ static NullableFlowState getResultState(BoundCollectionExpression node, Collecti
             }
         }
 
+        public override BoundNode? VisitCollectionExpressionSpreadElement(BoundCollectionExpressionSpreadElement node)
+        {
+            VisitRvalue(node.Expression);
```

Comment:
It looks like a case like `var x = [..(int[]?)null]` is handled by the `VisitForEachExpression` itself? Since we would be visiting a loop like `foreach (var elem in (int[]?)null) { ... }`?

Reply:
Yes, the top-level nullability of the expression being spread is handled in `VisitForEachExpression()`. For an example test, see `CollectionExpressionTests.SpreadNullability`.